pkgs/by-name/sk/skia-aseprite/package.nix

```diff
@@ -0,0 +1,116 @@
+{
+  aseprite,
+  clangStdenv,
+  expat,
+  fetchFromGitHub,
+  fetchgit,
+  fontconfig,
+  gitUpdater,
+  gn,
+  harfbuzzFull,
+  icu,
+  lib,
+  libglvnd,
+  libjpeg,
+  libpng,
+  libwebp,
+  libX11,
+  mesa,
+  ninja,
+  python3,
+  zlib,
+}:
+
+let
+  # deps.nix is generated by ./update.sh
+  depSrcs = import ./deps.nix { inherit fetchgit; };
+in
+clangStdenv.mkDerivation (finalAttrs: {
+  pname = "skia-aseprite";
+  version = "m102-861e4743af";
+
+  src = fetchFromGitHub {
+    owner = "aseprite";
+    repo = "skia";
+    rev = finalAttrs.version;
+    hash = "sha256-IlZbalmHl549uDUfPG8hlzub8TLWhG0EsV6HVAPdsl0=";
+  };
+
+  nativeBuildInputs = [
+    gn
+    ninja
+    python3
+  ];
+
+  preConfigure = with depSrcs; ''
+    mkdir -p third_party/externals
+    ln -s ${angle2} third_party/externals/angle2
+    ln -s ${dng_sdk} third_party/externals/dng_sdk
+    ln -s ${piex} third_party/externals/piex
+    ln -s ${sfntly} third_party/externals/sfntly
+  '';
+
+  configurePhase = ''
+    runHook preConfigure
```

Comment:
FYI: using `postConfigure` avoids having to add the hooks.

Reply:
I think that these fit the definition of a `configurePhase` and `buildPhase`, so they make sense to me

Reply:
I don't understand what you mean.

If you use `configurePhase` you have to additionally add the hooks. If you don't, you save that, and just add what you want to be executed. My point is to give you a shorter answer, such that you have less trouble/typing. I'm not saying what you proposed is wrong. The hooks are there. It works.

To be clear, I'm not saying your code is invalid. Just that, there is an easier to do that without having to overwrite the hooks.

This is not blocking your PR. I'm informing you of a better way of doing something.

Reply:
No I get what you mean.
What I mean is, a command like `gn gen` semantically makes sense to be the `configurePhase`, because it conceptually does the same thing that a `./configure` does.
Same with `ninja` - it's a build command, so putting it in `postBuild` could be confusing.
Is that clearer?

Reply:
I see, you are actually replacing the `configure`. Sorry.